### ISSUE TITLE
improve logging

### ISSUE THREAD

**Issue by mgrauer** [girder/girder#2741]
Move from our own logging to standard Python logging, which will make our instances easier to maintain and debug under WSGI.

**Comment by zachmullen** [comment]
@mgrauer could you clarify this a bit? What is non-standard about our logging that is problematic on WSGI?

**Comment by manthey** [comment]
In the current Girder default, we capture stdout and stderr to logs (as well as letting them go to stdout and stderr).  By default, stdout gets logged at "INFO" level and stderr gets logged at "ERROR" level.  

I like to see all of Girder logs in one place, rather than split between error and info logs.  We have a setting for our standard logging to set the level that goes to the "info" log:
```
[logging]
log_max_info_level = "CRITICAL"  # send everything to the info log
```

Note that this doesn't stop ERROR-level messages from going to our default error log (but you can point the error and info logs to the same file to change that).

You can prevent stdout and stderr from reaching the actual stdout and stderr by adding:
```
log_quiet = True
```

I've noticed that when multiple processes are writing to the same log file, nothing bad happens except when log file rotation occurs, at which point each process will often perform the rotation (even though only one should do so). 

We could (as a plugin or as part of core), add additional log handlers.  For instance, we could add a log handler that would use syslog.  If we did this in lieu of the file log handler, it would be a way to better handle multiple processes.

**Comment by zachmullen** [comment]
The problem here is that `_setupLogger` is called as an import side effect when importing anything under the girder package. `_setupLogger` adds these handlers that are problematic in the wsgi entry point.

Proposed solution: the import side effect should simply be `logger = logging.getLogger('girder')` rather than `logger = _setupLogger()`. It should be up to each entrypoint to configure the handlers, so the call to handler configuration could be moved to `girder.cli.serve:main`, and then wsgi could have its own call. That's certainly not the only way to fix it, we could alternatively keep the present behavior and just make the handlers more configurable via the config files.

**Comment by brianhelba** [comment]
@manthey I agree, we should have one log file with everything, as this is far easier to capture and analyze using any sort of automated tools (it's much easier to filter one file than merge two). I think this is a change we can make to Girder right now, and hopefully it may allow us to eliminate some lines of code and associated config fields.

@zachmullen The change to call `_setupLogger` as part of the CLI startup sounds great, and would help to clarify the fact that the *only* responsibility of `_setupLogger` is to attach a set of handlers to the central logger (accordingly, [this line](https://github.com/girder/girder/blob/master/girder/__init__.py#L168)
```logger.setLevel(logging.DEBUG if level is None else level)```
should be eliminated, since it changes the global state of the whole logger, not a handler).

I realize it's just an alternative, but I'd give a strong -1 to adding yet more configuration settings; indeed, I hope we can eliminate some with these changes. IMO, it's really hard to predict in advance exactly what consumers of your logs really want, and advanced users are almost always going to want to attach their own handlers and associated filters via standard Python logging mechanisms (which are globally-understood and totally flexible) rather than config file settings (which are Girder-specific and add lots of complexity to the codebase).

**Comment by brianhelba** [comment]
As part of this issue, I think [`logprint`](https://github.com/girder/girder/blob/master/girder/__init__.py#L236) should be removed entirely as well. Colored logging is not worth the additional non-standard complexity, and all messages should be going to the central logger anyway; any output to stdout should come exclusively via a handler attached to the central logger.

**Comment by brianhelba** [comment]
Hopefully, when these changes are complete, we can also have a guarantee that Girder (ignoring subprocesses) will not emit anything to `stdout` or `stderr` when run normally (since it's often hard to access and confusing / duplicative to see daemons write to these file descriptors). Of course, as a convenience, I could foresee a feature where the CLI would attach a handler to `stdout` / `stderr` when Girder is run in development mode from a TTY.

**Comment by manthey** [comment]
@brianhelba Removing `logprint` is a separable problem and should be a different issue. I understand the desire to make everything perfect all at once, but I'd recommend that PRs be allowed to be small and self-contained.  If we insist that an issue must change many things, then it makes it less likely someone will choose to work on, as it is harder to write.  Similarly, it harder to review, and reviews then tend to be less careful.

**Comment by manthey** [comment]
Logging has changed in girder 5.
